src/Query/ADQL_parser/ADQL_parser/init_factor.cxx

```diff
@@ -148,7 +148,12 @@ void ADQL_parser::init_factor() {
 
     cast_as %= ascii::no_case[ascii::string("NUMERIC")] |
                ascii::no_case[ascii::string("FLOAT4")] |
-               ascii::no_case[ascii::string("FLOAT8")];
+               ascii::no_case[ascii::string("FLOAT8")] |
+               ascii::no_case[ascii::string("VARCHAR2")] |
+               ascii::no_case[ascii::string("VARCHAR(40)")] |
+               ascii::no_case[ascii::string("VARCHAR")] |
```

Comment:
Curiously, does ADQL lib needs to be modified when cast to VACHAR(N) where N is not equals to 40?

Reply:
Yes.  I'll drop varchar(40) for now and add something more flexible when the need arises.